### PR TITLE
GitHub actions

### PR DESCRIPTION
**Pull request by endurance21**
added script for github action to perform build checks on every pull reques to master branch

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -25,5 +25,6 @@ jobs:
       uses: actions/setup-node@v1
       with:
         node-version: ${{ matrix.node-version }}
-    - run: npm ci
-    - run: npm run build --if-present
+    - run: npm install -g gatsby-cli
```
Comment (ryuk156) [inline .github/workflows/node.js.yml:28]: I think using yarn will be better @endurance21

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -25,5 +25,6 @@ jobs:
...
+    - run: yarn install 
+    - run: npm run build
```
Comment (ryuk156) [inline .github/workflows/node.js.yml:30]: even here you can use "yarn run build"

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (endurance21) [reply .github/workflows/node.js.yml:28]: does not really matter much , as those installed packages are gone once tasks are done on the Vm of github servers .

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 14.x, 15.x]
```
Comment (keturn) [inline .github/workflows/node.js.yml:19]: That's a broad version range. Do we need that, or can this project pick one version to build on?

Comment (pollend) [reply .github/workflows/node.js.yml:19]: this is just for publishing the site so picking one that works the best would be good. something not too old or too new I suppose :?

Comment (skaldarnar) [reply .github/workflows/node.js.yml:19]: 14 is the current LTS branch, so let's pick that? https://nodejs.org/en/about/releases/

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (skaldarnar) [reply .github/workflows/node.js.yml:28]: it does matter, as `npm install` will look for `package-lock.json` (https://docs.npmjs.com/cli/v6/configuring-npm/package-lock-json) while `yarn install` uses `yarn.lock`. As we have the latter checked in, we should use `yarn install` in the CI.

Comment (skaldarnar) [reply .github/workflows/node.js.yml:28]: And why do we install `gatsby-cli` globally? What is that used for, and why is not a dependency of this project?

File: .github/workflows/node.js.yml (hunk 2 above)
Comment (skaldarnar) [reply .github/workflows/node.js.yml:30]: indeed we should use the same tool for installing dependencies and actually building the site.

**Review by skaldarnar — CHANGES_REQUESTED** [review]
left bunch of comments before I realized that I'm not in Github's review mode...

File: .github/workflows/node.js.yml (hunk 3 above)
Comment (endurance21) [reply .github/workflows/node.js.yml:19]: Ya , I think 14 would be enought for our purpose.

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (endurance21) [reply .github/workflows/node.js.yml:28]: Actually we assume Gatsby-cli to be a independent tool.
And we can easily run it from any path.

Also if we use `gatsby-cli`  as dependency , we would a task runner like `npx` to run Gatsby like 
` npx Gatsby build ` or smth.

Comment (endurance21) [reply .github/workflows/node.js.yml:28]: > it does matter, as `npm install` will look for `package-lock.json` (https://docs.npmjs.com/cli/v6/configuring-npm/package-lock-json) while `yarn install` uses `yarn.lock`. As we have the latter checked in, we should use `yarn install` in the CI.
You are right 
In the context of working they both are differrent.

What I meant is we can use either npm or yarn , it should not matter much on testing build process thing.

**Comment by jdrueckert** [comment]
@endurance21 Hey there, would you still be interested in getting this ready?
If not, could you take over @ryuk156 ?

**Pull request closed without merging**